source/material_model/rheology/diffusion_creep.cc

```diff
@@ -91,11 +106,41 @@ namespace aspect
         // A: prefactor,
         // d: grain size, m: grain size exponent, E: activation energy, P: pressure,
         // V; activation volume, R: gas constant, T: temperature.
-        double viscosity_diffusion = 0.5 / p.prefactor *
-                                     std::exp((p.activation_energy +
-                                               pressure*p.activation_volume)/
-                                              (constants::gas_constant*temperature)) *
-                                     std::pow(grain_size, p.grain_size_exponent);
+	//
+	// If use_water_fugacity_diffusion = true this equation is modified:
+	//    viscosity = 0.5 * A'^(-1) * d^(m) * C_H2O^(-r) * exp((E' + P*V')/(RT))
+	// A' = A / A_H2O^(r)
+	// E' = E - r*E_H2O
+	// V' = V - r*V_H2O
```

Comment:
Now that I see this equation written down, most of the terms are pointless. A', E' and V' are all constants, so why not just require the user to put those in the prm file rather than make ASPECT calculate them billions of times?

I think the only term that needs to be in here is C_H2O^(-r), which is the standard term found in Mei and Kohlstedt (https://agupubs.onlinelibrary.wiley.com/doi/10.1029/2000JB900179), among others.

Reply:
That would mean that you could just add a single new creep parameter (creep_parameters.water_fugacity_exponent), rather than four.

Reply:
Sorry, in the cold light of day my comment from last night seems overly harsh. I could potentially see modifications to A, E and V being necessary, but perhaps via a relationship with `C_H2O`, rather than `r`? If you have a reference for this formulation, that would be great :)

Reply:
No need to apologize I think you make a great point! The equations could only be modified to include the water concentration and if you just calculate the reduction in the creep parameters  before hand then only the water_fugacity_exponent would need to be defined and that would save a lot of lines of code. I don't have a formulation for modifying A, E or V with variable C_H2O but I'll see what I can dig up!

Reply:
I didn't find a formulation for how to vary the creep parameters with varying C_H2O so I just simplified the existing code and included the same thing for dislocation creep, if you do know or come across something where A, E and V can be modified with C_H2O let me know! That would be very cool to try to implement

Reply:
@bobmyhill - Are you ok with the current approach (user modifies values in PRM file), and then modify the flow law parameters automatically later on if we find a robust relationship with C_H20?